numpy/core/src/multiarray/einsum_sumprod.c.src

```diff
@@ -1542,6 +1542,113 @@ bool_sum_of_products_outstride0_@noplabel@(int nop, char **dataptr,
 #  endif
 }
 
+/* For now, a catch all function for objects */
+static void
+object_sum_of_products_@noplabel@(int nop, char **dataptr,
```

Comment:
Is this code duplicated below?

Reply:
Ah, yes, 
the first represents `object_sum_of_products_<one, two, three, any>` the second is for the rest of the functions. 
In future commits I can place this all in one block of code...

Reply:
I'd recommend merging the blocks of code before attempting any of the other suggestions - it will make review much easier for us.

Reply:
Oh, sorry

I just put all these last few suggestions into one commit...

Reply:
That's fine too, as long as your new commit also includes the de-duplication